src/internal/connector/exchange/service_functions.go

```diff
@@ -86,44 +86,62 @@ func PopulateExchangeContainerResolver(
 }
 
 // Returns true if the container passes the scope comparison and should be included.
-// Also returns the path representing the directory.
+// Returns:
+// - the path representing the directory as it should be stored in the repository.
+// - the human-readable path using display names.
+// - true if the path passes the scope comparison.
 func includeContainer(
```

Comment:
actually, I think this function needs a good once-over. It's confusing because we have both `loc` and `locPath` but `locPath` is based off the path pulled from the container resolver. `loc` also does not seem to be returned at all so even if the container resolver has it we're not making use of it

Reply:
Not entirely sure what you're looking for with a refactor here.  Things are functional (outside of having had a bug on 121, and setting the locPath to the pb path).  And we need all the component parts.

Reply:
mm, I think I'd rather keep locPath nil throughout the whole passdown at this time.  Theoretically, in the future, if/when we expand this design to all kopia storage, loc will _never_ be nil.  In the meantime, if its nil, then it should be considered unsupported for that use case.  This will also help avoid "I didn't realize I wasn't populating it" gotchas.

That said, for purposes of directory comparison, just in this func, we already are falling back to the pb directory if loc is nil.

Reply:
> for purposes of directory comparison, just in this func, we already are falling back to the pb directory if loc is nil.

Are we though? all comparisons are done against `directory` which is set to `locPath.Folder()` on 138. `locPath` is only set if `loc != nil`

Reply:
Ugh.  Apparently I have more than one branch touching this code.  It's fixed _elsewhere_ ☹️ .  I'll make a change here, too.

Reply:
ah that would explain the confusion. Honestly you're doing great managing all the branches that are out right now